### DISCUSSION TITLE
[Helm Chart] Please provide an existingClaim feature (and make PVCs optional)

### DISCUSSION THREAD

**Discussion by PatrickHuetter** [nextcloud/all-in-one#4037] [Ideas] [answered]
Thanks for this interesting nextcloud-AIO project. I'm considering switching from my own helm chart deployment with the library/nextcloud docker image from Docker Hub to this one.

Currently that won't be possible easily since the PVCs are hardcoded within the helm chart contained in this repository and i would be dependent on the storage engine that needs to create and bind the PVCs. 

So i suggest to have the possibility to configure an existing claim for all services that have persistence like it's done in many other helm charts. Often via `persistence.existingClaim: some-pvc-name` values directive. That's also important in cases of backup and disaster recovery since the storage engine handles backups and volume snapshots transparently. So theoretically it's no disadvantage that BorgBackup is missing in the helm chart deployment since there are better options.

**Comment by szaimen** [comment, marked as answer]
Hi, thanks for the idea! 
However currently we don't have any plans to make them optional as more config options make things more unreliable and thus more prone to failure and instability. 
Also I don't understand why creating new PVCs is problematic in your case?
